### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Dorcas126**
In this project, I will use transactions to update and delete records. I will also use knowledge of aggregate functions and GROUP BY to answer analytical questions.

# Project Description
- Add a column species of type string to the animal's table. Modify schema.sql file.
- Insert the following data:
   - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Inside a transaction update the animal's table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
    - Update the animal's table by setting the species column to digimon for all animals that have a name ending in mon.
    - Update the animal's table by setting the species column to Pokemon for all animals that don't have species already set.
    - Verify that changes were made.
    - Commit the transaction.
    - Verify that changes persist after commit.
- Take a screenshot of the results of actions.
- Inside a transaction delete all records in the animal's table, then roll back the transaction.
- After the rollback verify if all records in the animal's table still exist. 
- Take a screenshot of the results of actions.
- Inside a transaction:
   - Delete all animals born after Jan 1st, 2022.
   - Create a savepoint for the transaction.
   - Update all animals' weight to be their weight multiplied by -1.
   - Rollback to the savepoint
   - Update all animals' weights that are negative to be their weight multiplied by -1.
   - Commit transaction
- Take a screenshot of the results of your actions.
- Wrote queries to answer the following questions:
   - How many animals are there?
   - How many animals have never tried to escape?
   - What is the average weight of animals?
   - Who escapes the most, neutered or not neutered animals?
   - What is the minimum and maximum weight of each type of animal?
   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- Take a screenshot of the results of queries.
- Add screenshots with the results of queries and with the results of play with transactions to the Pull Request description.

image

image

image

image

image

image

image

image

image

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Dorcas126,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Readme file professional ✔️
- Data added ✔️
- Schema added ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [ ] I noticed your queries are not according to the requirement, you did not add `SET species = 'unspecified'` but I saw that in the image below.
image

- [ ] I also noticed the `UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';`
image

- [ ] I will encourage you to show the results, at the moment, nothing is shown as result 
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 1]
```diff
@@ -55,4 +55,89 @@ animals
...
+UPDATE 
+animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+-- Update animals with no species set
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
```
Comment (TadesseDev) [inline queries.sql:70]: - [ ] As mentioned by the previous reviewer your queries are not according to the requirement, here the requirement states to `UPDATE` the animal's table by setting the species column to unspecified, then roll back the change. Here you are explicitly asked to verify that the change was made and rollbacked. Please add the `SELECT` statement before and after the change is rollback.

**Review by TadesseDev — APPROVED** [review]
x

File: queries.sql [hunk 2]
```diff
@@ -55,4 +55,89 @@ animals
...
\ No newline at end of file
+ weight_kg >=10.4 AND weight_kg <=17.3;
+
+-- Update animals with names ending in 'mon'
+UPDATE 
+animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+-- Update animals with no species set
```
Comment (TadesseDev) [inline queries.sql]: - [ ] As mentioned by the previous reviewer your queries are not according to the requirement, here the requirement states to `UPDATE` the animal's table by setting the species column to unspecified, then roll back the change. Here you are explicitly asked to verify that the change was made and rollbacked. Please add the `SELECT` statement before and after the change is rollback.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Dorcas126 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Dorcas126  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Made changes as requested by the previous reviewer :+1: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**